### PR TITLE
fix: add MANIFEST.in for SDist generation

### PR DESCRIPTION
**Pull request by henryiii**
This is missing. Mostly generated automatically by running `pipx run check-manifest -c`, then minor touchup to make `pipx run check-manifest` pass. (I used embarrassingly used check-manifest instead of check-sdist, which is my tool that works with all backends, due to the fact it can generate a MANIFEST.in due to it's setuptools integration and I'm lazy. Don't tell anyone ;) ).

This fixes `pipx run build`, and building SDists and wheels from SDists in general.

This makes the code saved in #618 even bigger. :D

### CODE REVIEW

**Comment by henryiii** [comment]
PS: I'd normally add check-manifest or check-sdist into your pre-commit (check-sdist is designed to be fast enough to do this!) to make sure it never misses anything, but you don't have one.

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,35 @@
+include *.txt
```
Comment (wjakob) [inline MANIFEST.in:1]: Could you change this to CMakeLists.txt?

Comment (henryiii) [reply MANIFEST.in:1]:
```suggestion
include CMakeLists.txt
```

Comment (henryiii) [reply MANIFEST.in]: Sure, done. Though in general, if you did add a something-else.txt file, I think I'd expect it in the SDist? .txt files usually aren't large, and generally SDists are supposed to basically be the same thing as git archives, unless you have large files.

**Comment by wjakob** [comment]
Thanks!

**Comment by henryiii** [comment]
Yes, this file can now be removed. It was only required to fix setuptools.

**Comment by wjakob** [comment]
Thanks. With this new tooling, what's the command to make a wheel for uploading to PyPI?

**Comment by henryiii** [comment]
Any way you used to do it should still work, except direct `setup.py` calls, but those have been deprecated for 4+ years and should in theory show warnings (it turns out it's really hard to make those calls show warnings without also trigging warnings in some valid ways, since they are internally used in setuptools plumbing).

My personal favorite is:

```bash
pipx run build
```

This will use [pipx](https://github.com/pypa/pipx) to download [build](https://github.com/pypa/build) into a temporary virtual environment and run it; by default, build makes an SDist and then builds a wheel from that SDist. If you run this again within a week, it will reuse the virtual environment. You can also pass `--installer=uv` to tell build to use `uv` instead of `pip` if you have `uv` installed, it's faster... by about the same amount of time it takes to type the extra characters. If you just want the wheel, use `--wheel`.

But any method works. You can `pip install build` and then use `python -m build`, for example. Or `pip wheel .` also works (doesn't make an SDist, though). GitHub Actions has pipx installed by default (manylinux does too), but there's also [hynek/build-and-inspect-python-package](https://github.com/hynek/build-and-inspect-python-package), which is highly optimized with uv and caching and also checks the SDist and wheel's created with some standard checks (and also uses build under the hood).

**Comment by henryiii** [comment]
Why are files like these gitignored?

```
  tests/test_classes_ext.pyi.ref
  tests/test_enum_ext.pyi.ref
  tests/test_functions_ext.pyi.ref
  tests/test_make_iterator_ext.pyi.ref
  tests/test_ndarray_ext.pyi.ref
  tests/test_stl_ext.pyi.ref
  tests/test_typing_ext.pyi.ref
```

**Comment by wjakob** [comment]
Good point, fixed.

**Pull request merged**